ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,375 @@
+          const messageCode = rawMessageText.substring(0, 2);
+          var messageText = rawMessageText.substring(2, rawMessageText.length);
+          var messageType = undefined;
+          var student = undefined;
+          messageCode === 'i]' ? messageType = 'info' : messageCode === 'u]' ? messageType = 'user' : messageType = 'student';
+          if(messageType === 'student'){
+            const splitStudentMessage = rawMessageText.split('s]');
+            const studentId = Number(splitStudentMessage[0]);
+            messageText = splitStudentMessage[1];
+            student = _.find(this.props.question.students, student => student.id===studentId);
+          }
+          const message = {
+            type: messageType,
+            text: messageText,
+            key: messageText,
+            student
+          };
+          messages.push(message);
+        }
+      }
+    }else{
+      const message = {
+        type: 'info',
+        text: questionObject.text,
+        key: questionObject.text
+      };
+      messages.push(message);
+    }
+    
+    return messages;
+  },
+  
+  componentDidMount() {
+    this.setInterval(this.updateTimer, ONE_SECOND);
+  },
+
+  updateTimer() {
+    if(!this.isReadyToMoveOn()){
+      if(this.state.elapsedMs === 90000){
+        this.addMessages([{type: 'info', text: 'Remember, these are supposed to be quick responses.'}]);
+      }
+      this.setState({ elapsedMs: this.state.elapsedMs + ONE_SECOND });
+    }
+  },
+  
+  render(){
+    const {scaffolding, question} = this.props;
+    const {helpType, shouldShowStudentCard} = scaffolding;
+    const {selectedStudent} = this.state;
+    return (
+      <div>
+        <div>
+          <div>
+            <div style={styles.textBody}>
+              <TextBody 
+                question={this.props.question}
+                messages={this.state.messages}
+                onOpenStudentDialog={this.onOpenStudentDialog}
+                onOpenInfoDialog={this.onOpenInfoDialog}
+                />
+            </div>
+          </div>
+        </div>
+        <div>
+          <div> 
+            <div style={styles.textFooter}>
+              <TextFooter 
+                onQuestionDone={this.onNextQuestion}
+                setInitialResponse={this.setInitialResponse}
+                setRevisedResponse={this.setRevisedResponse}
+                setPassedResponse={this.setPassedResponse}
+                log={{logResponse: this.logResponse, logRevision: this.logRevision, logRevisionDeclined: this.logRevisionDeclined}}
+                isReadyToMoveOn={this.isReadyToMoveOn}
+                helpType={helpType}
+                elapsedMs={this.state.elapsedMs}
+                showExample={this.showExample}
+                nextExample={this.nextExample}
+                nextButtonLabel={this.props.isLastQuestion ? 'Finish' : 'Next Question'}
+                />
+            </div>
+            <div>
+              {selectedStudent !== undefined &&
```

Comment:
This looks good, but I'd recommend splitting out the rendering of the dialogs to their own method.